custom-routes-vis/src/main/java/com/google/sps/servlets/GetDirectionsServlet.java

```diff
@@ -81,14 +83,18 @@ public URLConnection sendGetRequestToDirectionsApi(String origin, String destina
    * @param destination Destination of the requested routes in "lat,lng" format.
    * @param alpha Whether to send the request to the alpha version of the API.
    * @param key The API key.
+   * @param rateCard The Rate Card for this call to the API.
    */
   public URLConnection sendPostRequestToRoutesApi(String origin, String destination, boolean alpha,
-      String apiKey) throws IOException, MalformedURLException {
+      String apiKey, String rateCard) throws IOException, MalformedURLException {
     URL routesUrl;
+    String rateCardString;
     if (alpha) {
       routesUrl = new URL("https://routespreferred.googleapis.com/v1alpha:computeCustomRoutes");
+      rateCardString = "\"routeObjective\": {\"rateCard\": " + rateCard + "},";
     } else {
       routesUrl = new URL("https://routespreferred.googleapis.com/v1:computeRoutes");
+      rateCardString = "";
```

Comment:
I think it would be empty by default, so you don't need this.

Reply:
maybe initialize it to ""?